container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1482,6 +1484,105 @@ func (manager *KubernetesManager) RunExecCommand(
 	return successExecCommandExitCode, nil
 }
 
+func (manager *KubernetesManager) RunExecCommandWithStreamedOutput(
+	ctx context.Context,
+	namespaceName string,
+	podName string,
+	containerName string,
+	command []string,
+) (chan string, chan *exec_result.ExecResult) {
+	execOutputChan := make(chan string)
+	finalExecResultChan := make(chan *exec_result.ExecResult)
+	go func() {
+		defer func() {
+			close(execOutputChan)
+			close(finalExecResultChan)
+		}()
+		execOptions := &apiv1.PodExecOptions{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "",
+				APIVersion: "",
+			},
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+			Container: containerName,
+			Command:   command,
+		}
+
+		//Create a RESTful command request.
+		request := manager.kubernetesClientSet.CoreV1().RESTClient().
```

Comment:
Not super familiar with Kubernetes API, I trust you that's the best way to run the exec command. Though it seems quite painful

Reply:
very painful lol. I followed how it was done for the initial `kubernetesManager.RunExecCommand`, but all examples I found online and api point to doing it this way :/